src/main/java/pl/javastart/lista/TaskRepository.java

```diff
@@ -0,0 +1,9 @@
+package pl.javastart.lista;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+
+public interface TaskRepository extends JpaRepository<Task, Long> {
+    List<Task> displayDoneTasks(boolean isDone);
+}
```

Comment:
Gdzieś tutaj wywala mi błąd:
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'taskRepository' defined in pl.javastart.lista.TaskRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Invocation of init method failed; nested exception is java.lang.IllegalArgumentException: Not a managed type: class pl.javastart.lista.Task
W sumie nie rozumiem dlaczego bo w zadaniu z bazami filmów jest ok a tu robię kropka w kropkę i nie jest ok ;/ 
W sumie też przez to nie mogę sprawdzić czy wgl mi działa baza danych i dlatego zatrzymałam się w połowie home.html i nie ruszyłam dalej

Reply:
Podejdziemy do tego etapowo ;) Pierwszy błąd jaki leci:
`Error creating bean with name 'taskRepository' defined in pl.javastart.lista.TaskRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Invocation of init method failed; nested exception is java.lang.IllegalArgumentException: Not a managed type: class pl.javastart.lista.Task`
mówi nam, że aby skorzystać z generowania repozytoriów przez Sprina, to w miejsce `Task` w deklaracji `public interface TaskRepository extends JpaRepository<Task, Long>` musi być przekazana klasa będąca encją w rozumieniu JPA. By klasa `Task` była takową, to musimy na nią nałożyć adnotację `@Entity`.

Po zrobieniu tego leci następujący błąd:
`Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Invocation of init method failed; nested exception is org.hibernate.AnnotationException: No identifier specified for entity: pl.javastart.lista.Task`. Czyli w naszej encji `Task` zabrakło nam oznaczenia pola, które jest identyfikatorem (każda encja musi mieć identyfikator). Tak więc należy dać adnotację `@Id` na pole `id` tejklasy ;) Btw dorzucić tutaj generowanie identfikatora (adnotacja `@GenerateValue`) :P 

Teraz jak uruchommimy aplikację to leci błąd:
`Error creating bean with name 'taskRepository' defined in pl.javastart.lista.TaskRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Invocation of init method failed; nested exception is org.springframework.data.repository.query.QueryCreationException: Could not create query for public abstract java.util.List pl.javastart.lista.TaskRepository.displayDoneTasks(boolean)! Reason: Failed to create query for method public abstract java.util.List pl.javastart.lista.TaskRepository.displayDoneTasks(boolean)! No property displayDoneTasks found for type Task!; nested exception is java.lang.IllegalArgumentException: Failed to create query for method public abstract java.util.List pl.javastart.lista.TaskRepository.displayDoneTasks(boolean)! No property displayDoneTasks found for type Task!`. Oznacza to, że Spring ma problem z wygerowaniem kodu metody `displayDoneTasks` na podstawie jej nazwy. Jednym z rozwiązań jest zmiana nazwy metody na taką, którą Spring zrozumie. Zwyczajowo konwencja jest taka: `findBy` + pole po którym szukamy + (opcjonalnie) `And` + kolejne pole po ktorym szukamy. Zatem w naszym wypadku metoda powinna się nazywać `findByDone(boolean isDone)` ;)
Jeśli jesteś ciekawa jakie są konwencje nazewnicze tych metod, to wiecej jest opisane tutaj:
https://docs.spring.io/spring-data/commons/docs/2.5.4/reference/html/#repository-query-keywords

Reply:
Btw widze ze w kontrolerze zostaly nazwy po filmach ;P

Reply:
Oj no wiem XD Jak już człowiek się zdenerwuje to szuka wszędzie pomocy XD